src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -208,7 +208,6 @@ export default class SourceFileLinter {
 		const className = node.name?.getText() ?? "<unknown>";
 		const rendererMember = node.members.find((member) => {
 			return (ts.isPropertyDeclaration(member) || ts.isMethodDeclaration(member)) &&
-				member.modifiers?.some((modifier) => modifier.kind === ts.SyntaxKind.StaticKeyword) &&
```

Comment:
My thinking was that we should only complain about a non-static `renderer` property when there is no static one. It should be allowed for controls to define a valid (static) renderer and use an instance property with the same name. Only when no static renderer is defined, an instance property IMO indicates a wrong pattern.

The `members.find()` will only find the first occurrence. I would therefore suggest to keep the old check and introduce a new check only in the case that there no `rendererMember` was found (around line 226 in the old code).

Reply:
Thanks!